lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -79,11 +79,18 @@ def workspace_did_change_watched_files(changes)
         return unless T.must(@global_state).enabled_feature?(:tapiocaAddon)
         return unless @rails_runner_client # Client is not ready
 
+        has_route_change = T.let(false, T::Boolean)
+
         constants = changes.flat_map do |change|
           path = URI(change[:uri]).to_standardized_path
           next if path.end_with?("_test.rb", "_spec.rb")
           next unless file_updated?(change, path)
 
+          if File.basename(path) == "routes.rb" || File.fnmatch?("**/routes/**/*.rb", path)
```

Comment:
Does this need `File::FNM_PATHNAME | File::FNM_EXTGLOB`?

Reply:
I don't believe so. We're not using the `{}` syntax and we're declaring all path parts we're interested in.

What scenario is this going to fail without those flags?

Reply:
I think we need `File::FNM_PATHNAME` but not the other:

```
irb(main):005> File.fnmatch?("**/routes/**/*.rb", "routes/foo.rb")
=> false
irb(main):006> File.fnmatch?("**/routes/**/*.rb", "routes/foo.rb", File::FNM_PATHNAME)
=> true
```

Reply:
Good catch.